src/view/components/common/modal-dialog/dialog-component.ts

```diff
@@ -49,6 +50,10 @@ export class ModalDialogComponent {
       } else {
         this.removeEscapeListener()
       }
+
+      if (!change.hidden.currentValue){
+        setTimeout( () => this.open.emit(false), 2);
```

Comment:
Please add a comment describing why this hack is required. Because it's very much a hack and we'll need to fix it the right way eventually.